content/tutorials/guides/how-to-get-families-and-attributes.md

```diff
@@ -0,0 +1,232 @@
+
+// Transform JSON response to array
+$data = json_decode($response->getBody()->getContents(), true);
+
+// Collect family codes from paginated API
+$codes = array_column($data['_embedded']['items'], 'code');
+while (array_key_exists('next', $data['_links'])) {
+    $response = $client->get(end($data['_links']['next']));
+    $data = json_decode($response->getBody()->getContents(), true);
+    $codes = array_merge($codes, array_column($data['_embedded']['items'], 'code'));
+}
+
+// Save family codes into storage
+saveFamilyCodes($codes);
+```
+
+Get attribute codes by requesting the PIM API
+
+```php [activate:PHP]
+const API_URL = '/api/rest/v1/families';
+
+// Make an authenticated call to the API
+$response = $client->get(API_URL);
+
+// Transform JSON response to array
+$data = json_decode($response->getBody()->getContents(), true);
+
+// Collect attribute codes from paginated API
+$attributes = array_unique(array_merge(...array_column($data['_embedded']['items'], 'attributes')));
+while (array_key_exists('next', $data['_links'])) {
+    $response = $client->get(end($data['_links']['next']));
+    $data = json_decode($response->getBody()->getContents(), true);
+    $attributes = array_unique(array_merge($attributes, ...array_column($data['_embedded']['items'], 'attributes')));
+}
+
+// Save attribute codes into storage
+saveAttributesCodes($attributes);
+```
+
+Store family codes in a <b>family_code_list</b> and attribute codes in a separate list (<b>attribute_code_list</b>). We will deal with <b>attribute_code_list</b> later in this tutorial.
+
+::: tips
+Warning! with the API call GET api/rest/v1/families, you will collect <b>all the families into the database</b>! Please ask yourself this question before continuing: <i>Do I really need all of them?</i> 
+At this step, it’s the perfect occasion to save time later, during products synchronization. We strongly advise you to <b>filter your families</b> as much as you can before building family_code_list and attribute_code_list.<br> 
+👉 One way to do this is <a href="/documentation/filter.html#by-family-codes" target="_blank" rel="noopener noreferrer">the family codes filter</a>
+:::
+
+### 2 - Collect family variants
+::: info
+This step is mandatory if you want to synchronize product variants later. If not, jump to the third step.
+:::
+
+Get family variants by requesting the PIM API for each families
+
+```php [activate:PHP]
+const API_URL = '/api/rest/v1/families/%s/variants';
+
+// Get family codes from storage
+$codes = getFamilyCodes();
+
+// Collect family variants from paginated API
+$variants = [];
+foreach ($codes as $code) {
+    $response = $client->get(sprintf(API_URL, $code));
+    $data = json_decode($response->getBody()->getContents(), true);
+    if (!empty($data['_embedded']['items'])) {
+        $variants = array_merge($variants, $data['_embedded']['items']);
+    }
+}
+
+// Save variants into storage
+saveVariants($variants);
+```
+::: tips
```

Comment:
You can remove this tip now, because you iterate over your families into an independent loop